src/client.ts

```diff
@@ -498,7 +498,7 @@ export class StreamClient<
       withRecentReactions?: boolean;
     } = {},
   ) {
-    if (options.enrich) {
+    if (Object.prototype.hasOwnProperty.call(options, 'enrich')) {
```

Comment:
that's odd, if enrich is not a prop of objects this should return undefined which is a falsey value. 

Do you know why this is not the case for you?
Did you try to use something stricter like `options.enrich === true`?
What do you get if you log `options` and `options.enrich`?

Reply:
@tbarbugli the fact that it returns a falsey is exactly the problem.

because this.enrichByDefault = !this.usingApiSecret; is true when on the client side.
There is no way to turn off enrichment.
normally you'd do
{ enrich: false }
But as you correctly pointed out a false value doesn't allow it to go into the if, and so it will be true.

Reply:
@tbarbugli this actually introduced bugs for anyone trying to use the other properties, the correct way is to only go with enrichByDefault if we haven't explicitly passed { enrich: false }

Reply:
what about checking for null?

```js
if (options.enrich != null) {
    // same as before use options.enrich
}

// rely on default || other params
```

Reply:
@tbarbugli 
consider the case where
enrichByDefault -> true
You pass
`{ enrich: false, ownReactions: true }`

`if (options.enrich != null) {`
means `shouldUseEnrichEndpoint` we will return false, when in my understanding it should return true.
At least that's what it does if enrichByDefault -> false.